### PR TITLE
Assert RAFT input resolution is 128 x 128 or higher

### PR DESCRIPTION
**Pull request by ChristophReich1996**
Hi torchvision community,

As described in issue #7338 the RAFT architecture is only suitable for input resolutions of `128 x 128` or higher (and res. divisible by 8). Otherwise, when performing [grid_sample](https://github.com/pytorch/vision/blob/01ef0a68b6ec00452391251fc16c38e58b92bf07/torchvision/models/optical_flow/_utils.py#L13) a division by zero occurs in the lowest resolution stage. Thus, the full output of the RAFT model will be `nan` for resolutions smaller than `128 x 128`.
This pull request just adds a check if the input images entail a resolution of `128 x 128` or higher to avoid `nan`s.

### CODE REVIEW

File: torchvision/models/optical_flow/raft.py [hunk 1]
```diff
@@ -475,6 +475,8 @@ def forward(self, image1, image2, num_flow_updates: int = 12):
             raise ValueError(f"input images should have the same shape, instead got ({h}, {w}) != {image2.shape[-2:]}")
         if not (h % 8 == 0) and (w % 8 == 0):
             raise ValueError(f"input image H and W should be divisible by 8, instead got {h} (h) and {w} (w)")
+        if not (h < 128) and (w < 128):
+            raise ValueError(f"input image H and W should be equal or larger than 128, instead got {h} (h) and {w} (w)")
```
Comment (NicolasHug) [inline torchvision/models/optical_flow/raft.py:479]: This is not going to raise an error if `h < 128` - I assume this is not what we want?

```suggestion
        if h < 128 or w < 128:
            raise ValueError(f"input image H and W should be equal to or larger than 128, instead got {h} (h) and {w} (w)")
```

**Review by NicolasHug — COMMENTED** [review]
Thanks for the PR @ChristophReich1996 , LGTM with a minor modification below

File: torchvision/models/optical_flow/raft.py (hunk 1 above)
Comment (ChristophReich1996) [reply torchvision/models/optical_flow/raft.py]: Resolved, thanks!

**Review by NicolasHug — APPROVED** [review]
Thanks a lot @ChristophReich1996 LGTM. let's just wait for the checks to be green before merging.

**Comment by ChristophReich1996** [comment]
The RAFT tests fail due to the low resolution in the tests. The issue (file size vs res.) is described here:
https://github.com/pytorch/vision/blob/b030e9363eab6089cc580725ee703cf2f01f3765/test/test_models.py#L1032-L1044

Can we go to 128 x 128 in this test?

**Comment by NicolasHug** [comment]
Ah, I remember that now... That's unfortunate. I guess this was the reason we didn't hard-code a check like the one proposed in this PR.

Our tests are pretty terrible overall and we've been thinking about revamping them for a long time now. I don't know when this will be done but unfortunately, I'm not sure we'll be able to change the size of the input for now. That would make the expected weights files too big, which would then increase the size of the repo.

Sorry @ChristophReich1996 , unfortunately we might not be able to merge that PR right now 😞

**Comment by ChristophReich1996** [comment]
**Idea:** Let's not check if the image is large enough but let's check if the feature maps are large enough for the correlation block... Since, as in the test, it also depents on the correlation block parameters if `nan`s occur. Let me know what you think!

https://github.com/pytorch/vision/blob/126e3e7527243fd838bf2929e6483326815283de/torchvision/models/optical_flow/raft.py#L484

File: torchvision/models/optical_flow/raft.py [hunk 2]
```diff
@@ -480,7 +480,16 @@ def forward(self, image1, image2, num_flow_updates: int = 12):
...
-
+        
+        _, _, h_fmap, w_fmap = fmap1.shape
+        if not (((h_fmap // 2**(self.corr_block.num_levels - 1))) < 2) and (
+                ((w_fmap // 2**(self.corr_block.num_levels - 1))) < 2
+        ):
+            min_res = 2 * 2**(self.corr_block.num_levels - 1) * 8
+            raise ValueError(
+                f"input image resolution is too small image resolution should be at least {min_res} (h) and {min_res} (w), got {h} (h) and {w} (w)"
+            )
```
Comment (NicolasHug) [inline torchvision/models/optical_flow/raft.py:491]: It should be equivalent to simply check `h_fmap` and  `w_fmap` against `min_res`? Something like:

```suggestion
        _, _, h_fmap, w_fmap = fmap1.shape
        min_res = 2 * 2**(self.corr_block.num_levels - 1) * 8
        if (h_fmap, w_fmap) < (min_res, min_res):
            raise ValueError(...)
```

**Review by NicolasHug — COMMENTED** [review]
Sorry for the late review @ChristophReich1996  and thanks for the changes. I made a comment below to hopefully simplify the computations. I'd also suggest to put those checks within the `corr_block.build_pyramid()` method, instead of in the `forward()` pass. It's functionally equivalent but it makes sense to put the pyramid-related checks within the pyramid class.

File: torchvision/models/optical_flow/raft.py (hunk 2 above)
Comment (ChristophReich1996) [reply torchvision/models/optical_flow/raft.py]: New version is ready :)

Comment (ChristophReich1996) [reply torchvision/models/optical_flow/raft.py]: This looks good to me, thanks!

**Comment by NicolasHug** [comment]
Thanks a lot for the PR @ChristophReich1996 !

**Pull request merged**